src/rules/no-untyped-mock-factory.ts

```diff
@@ -0,0 +1,95 @@
+import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/utils';
+import { createRule, isTypeOfJestFnCall } from './utils';
+
+const hasReturnType = (
+  node:
+    | TSESTree.FunctionExpression
+    | TSESTree.ArrowFunctionExpression
+    | TSESTree.Node,
+): boolean => {
+  if (
+    node.type === AST_NODE_TYPES.FunctionExpression ||
+    node.type === AST_NODE_TYPES.ArrowFunctionExpression
+  ) {
```

Comment:
Let's use our `isFunction` utility here (tbh I'd probably inline this whole function since we only do this check in one place, but up to you)